### DISCUSSION TITLE
Advertise local IP and allow a Direct IP Connection when devices are on the same network

### DISCUSSION THREAD

**Discussion by kozman1000** [rustdesk/rustdesk#8654] [General]
When the feature in Settings > Security menu > Security > "Enable direct IP access" (@ port 21118) was introduced, this offered a secondary method of accessing a PC on a local network when, perhaps, the public Rustdesk network / relay wasn't accessible (i.e., during DDoS attack) but the PC was on the same local network even if local discovery may be turned off but "Enable direct IP access" is enabled on a remote client. I think some more blood can be squeezed from this stone, so to speak.

Imagine this scenario: A remote user wants me to access their PC but the remote PC cannot access or is being blocked from the public Rustdesk network / relays because of something like https://github.com/rustdesk/rustdesk/discussions/7008 or some such similar reason. The remote user may also have enabled "Deny lan discovery" but if I happen to have VPN access to this user's local network, I can VPN to their network and now I can reach them if they have enabled "Enable direct IP access". But what's their local IP address? Rustdesk doesn't overtly advertise this and I would have to guess or ask the user to do an IPCONFIG /ALL (or find the IP via connection Properties menu) to get me their IP address.

Theoretically, on the "Discovered" tab, I should likely be able to see this user's remote device now that I am VPN'ed into their local network (unless they've enabled "Deny lan discovery"). But if "Enable direct IP access" is enabled on their remote client, there's hope to connect to this device. It would be nice to see their local network IP advertised somewhere and / or an option in the three vertical dots menu for this user's session as is shown in the picture below.

![allow-direct-IP-connection-when-enabled-same-network](https://github.com/rustdesk/rustdesk/assets/139574773/4460f28b-9dd7-4f8e-8893-faba4d31ed45)

**Comment by daihuiao** [comment]
It is necessary to directly connect local devices. It is ridiculous that my LAN control device needs to be connected to the server.

**Comment by BasRuggeberg** [comment]
How can I enable direct IP access in the rustdesk app on my iPad?

**Comment by DogChina** [comment]
<img width="210" height="354" alt="0260909212624" src="https://github.com/user-attachments/assets/8bb1d426-ba9c-4465-a881-ed709c29e91d" />
<img width="225" height="384" alt="60909210639" src="https://github.com/user-attachments/assets/b5a439ba-0b32-47f3-9a91-d712536f153e" />

**Reply by DogChina** [reply]
https://github.com/DogChina/rustdesk-costom/releases/tag/1.4.9
